python/setup.py

```diff
@@ -1,9 +1,56 @@
 from setuptools import setup
+from glob import glob
+import shutil
+import sys
+import os
 
+# single source of truth for package version
+version_ns = {}
+with open(os.path.join("lolopy", "version.py")) as f:
+    exec(f.read(), version_ns)
```

Comment:
I'm not familiar with this pattern; why are you using `exec`?

Reply:
I was following one of these [patterns for accessing the library version from `setup.py`](https://packaging.python.org/guides/single-sourcing-package-version/).

If `exec` makes you nervous, there are other routes (e.g., extracting the version number with a regex) .

Reply:
No, the pattern is fine.  Thanks for the link.